app/forms/idv/finance_form.rb

```diff
@@ -39,6 +39,13 @@ def finance_type_valid
       end
     end
 
+    def ccn_valid
+      return true unless finance_type == :ccn
+      unless finance_account =~ /^\d{8}$/
+        errors.add :finance_account, I18n.t('idv.errors.invalid_ccn')
+      end
+    end
```

Comment:
Ah that's better. I had tried passing `message` in various ways and got all kinds of errors. Your example is much better than the Rails docs. Pushed in 292f6de